arangod/Aql/RestAqlHandler.cpp

```diff
@@ -336,7 +336,7 @@ bool RestAqlHandler::registerSnippets(
         // No need to cleanup...
       }
 
-      _queryRegistry->insert(qId, query.get(), ttl, true);
+      _queryRegistry->insert(qId, query.get(), ttl, true, false);
```

Comment:
Can this be changed so that `std::move(query)` is passed to QueryRegistry::insert instead of passing the raw pointer and then releasing it somewhere below?

Reply:
I think the question is if we need two variants of the `insert` method, one which does not keep the lease (like here), and another one which does keep the lease. So far, I have only seen calls with `false`. If we need the two variants, we either need two methods or a switch as implemented. As far as I see, we cannot sometimes move and sometimes not. And the method cannot distinguish, I guess.